src/transactions/feed/TransactionDetailsScreen.test.tsx

```diff
@@ -313,6 +324,58 @@ describe('TransactionDetailsScreen', () => {
     expect(getByText('transactionDetailsActions.checkPendingTransactionStatus')).toBeTruthy()
   })
 
+  it(`renders check status action for pending ${TokenTransactionTypeV2.Approval} transacton`, () => {
+    const { getByText } = renderScreen({
+      transaction: approvalTransaction({
+        status: TransactionStatus.Pending,
+      }),
+    })
+
+    expect(getByText('transactionDetailsActions.checkPendingTransactionStatus')).toBeTruthy()
+  })
+
+  it(`renders the correct details for ${TokenTransactionTypeV2.Approval} transacton`, () => {
+    const { getByText, getByTestId } = renderScreen({
+      transaction: approvalTransaction({
+        status: TransactionStatus.Complete,
+      }),
+      storeOverrides: {
+        tokens: {
+          tokenBalances: {
+            'ethereum-sepolia:0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': {
```

Comment:
Do we need this token balance?

Reply:
yes, although technically i only need this object in order to get the token symbol for the message (i.e. you've approved USDC). i can remove some unnecessary properties

Reply:
does this `tokenBalances` object always have all supported tokens, even if the balance is 0?

because what if we had an approval for a token sometimes earlier, but then drained its balance? can we rely that it will be there?

Reply:
good question, yes the token info will be there even if the balance is zero. we fetch the token balances from the getTokenInfo CF and combine the last known balance with it before storing it all in redux https://github.com/valora-inc/wallet/blob/2d551d94a99955a93564fc9792d7a0e033dda9d9/src/tokens/saga.ts#L195-L207